subsys/bluetooth/host/att.c

```diff
@@ -188,6 +188,15 @@ static int chan_send(struct bt_att_chan *chan, struct net_buf *buf,
 			return -EAGAIN;
 		}
 
+		/* Check the encryption level for EATT */
+		if (IS_ENABLED(CONFIG_BT_EATT) &&
+		    chan->att->conn->sec_level < BT_SECURITY_L2) {
+			/* Vol 3, Part G, Section 5.3.2 Channel Requirements states:
+			 * The channel shall be encrypted.
+			 */
+			return -EINVAL;
+		}
```

Comment:
This is only for EATT channels. Checking `IS_ENABLED(CONFIG_BT_EATT)` does not verify that the channel is an enhanced ATT bearer, and this check will apply for the unenhanced ATT channel which is a fixed L2CAP channel. 

```suggestion
		if (IS_ENABLED(CONFIG_BT_EATT) &&
		    atomic_test_bit(chan->flags, ATT_ENHANCED)
		    chan->att->conn->sec_level < BT_SECURITY_L2) {
			/* Vol 3, Part G, Section 5.3.2 Channel Requirements states:
			 * The channel shall be encrypted.
			 */
			return -EINVAL;
		}
```

I also wonder if the check should check against `CONFIG_BT_EATT_SEC_LEVEL` instead of `BT_SECURITY_L2`, but I am a bit unsure about that one. @hermabe @Vudentz do you know?

Reply:
Right so it should just be 

```suggestion
		if (chan->att->conn->sec_level < BT_SECURITY_L2) {
			/* Vol 3, Part G, Section 5.3.2 Channel Requirements states:
			 * The channel shall be encrypted.
			 */
			return -EINVAL;
		}
```

Reply:
Yes. Should also use the `bt_conn_get_security()` function so that it compiles if `BT_SMP` is not set and `conn->sec_level` does not exist.